provider/pkg/provider/provider.go

```diff
@@ -1368,6 +1369,23 @@ func (k *kubeProvider) Diff(
 		}
 	}
 
+	if metadata.ReplaceUnready(newInputs) {
+		switch newInputs.GetKind() {
+		case "Job":
+			jobChecker := states.NewJobChecker()
+			job, err := clients.FromUnstructured(newInputs)
+			if err == nil {
+				jobChecker.Update(job)
```

Comment:
Is it okay to do this wait operation from Diff?  Do we normally do that?

Reply:
It's not actually running the full await logic; this is a point in time check based on the current status of the Job.

I'm not sure if there's precedent for that, but it shouldn't have a noticeable effect on performance, and Diff is the only place where we can indicate a replacement is required.